torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -635,6 +635,10 @@ class TORCH_API ProcessGroupNCCL : public Backend {
   std::unordered_map<std::string, std::vector<std::shared_ptr<NCCLComm>>>
       devNCCLCommMap_;
 
+  // The NCCL communicators currently in progress of being initialized.
+  std::unordered_map<std::string, std::vector<std::shared_ptr<NCCLComm>>>
+      inProgressCommMap_;
```

Comment:
Minor nit: Is it more correct to call this something like `inInitialization` or `duringInitialization`, as any comms in `devNCCLCommMap_` can also have an in-progress status?